### ISSUE TITLE
Don't use namespacing in variable names

### ISSUE THREAD

**Issue by ralt** [vsedach/Parenscript#14]
Hi,

The issue with using namespacing in variable names is that it will namespace everything, including any mention I make to `document`, `window`, etc. For example:

``` lisp
(chain document (query-selector-all selector))
```

Will output:

```
_prefix_document._prefix_querySelectorAll(_prefix_selector);
```

Which is not really what I want.

Any idea of how I can do that?

The thing is, I want to include the javascript in many web pages, so I cannot afford to _not_ have namespaces. I also want to have several files, so I need the packaging system :-)

The solution I'm using for now is something like this:

``` lisp
(setf (getprop window '*foo*) (or (getprop window '*foo*) (create)))
(setf (getprop (getprop window '*bar*) 'bar) #'bar) ;; export FOO.bar
```

Basically, using js "namespacing" manually. I have to wrap this in an IIFE manually when using `ps:compile-file` though, so my code ends up like this:

``` lisp
(defparameter *foo-ps* (concatenate
                   'string
                   ";(function(){"
                   (ps:ps-compile-file
                     (asdf:system-relative-pathname :bar "foo.ps"))
                   "}());"))
```

To have the js code available.

I'd love a better solution, but I don't see it yet.

Cheers,
Florian

**Comment by vsedach** [comment]
Hi Florian,

Symbol obfuscation is per-package, and there is a package called
#:ps-dhtml-symbols
that defines the standard and de-facto standard DOM identifiers:
https://github.com/vsedach/Parenscript/blob/master/src/js-dom-symbol-exports.lisp#L1071

If you :USE that package in your package definition, that will keep those
symbols from being mangled.

As a note to myself, TODO need to add HTML5 symbols to that file.

Happy hacking,
Vladimir

**Comment by Ralt** [comment]
Hi @vsedach,

Thanks for your feedback!

Indeed, `#:ps-dhtml-symbols` looks like what I want!

I'm using this fix until you can add the missing DOM standards :-)

```lisp
;; fix parenscript missing DOM methods in #:ps-dhtml-symbols
;; See https://github.com/vsedach/Parenscript/issues/14
(defpackage #:lispkit.dhtml-missing-methods
  (:documentation "Missing functions used in lispkit source code.")
  (:export
   #:query-selector-all
   #:inner-width
   #:inner-height
   #:get-bounding-client-rect
   #:text-content))

(ps::re-export-symbols '#:lispkit.dhtml-missing-methods '#:ps-dhtml-symbols)
```

This way, I can use the obfuscation technique without any worry!

Good luck when adding the new symbols, I'm afraid there are a lot of them...

Cheers,
Florian

**Comment by jasom** [comment]
@ralt do you have a link to the definitive list of symbols?  I'll be happy to add them if so.

**Comment by ralt** [comment]
@jasom you will have to find all the DOM3 and DOM4 new symbols. I'm not sure it's easy to find such a list...
